### ISSUE TITLE
Run a job right away from foreground?

### ISSUE THREAD

**Issue by cristianocca** [vikeri/react-native-background-job#99]
Is it possible from the app (react) to run a registered job right away as a headless job? In order to avoid waiting for it to be picked up.

Use case is this: An uploader queue that will grab stuff from the database and upload to a server over time. If the user has the app opened it's ideal that the uploads happen right away but are also not interrupted on app pause (i.e, going to another app).

Right now, if a job is scheduled you will still need to wait for the scheduler to pick it up over time (every 15 minutes).

Simply running the upload code as a regular function is not viable since app suspend can stop the whole process and it looks like code running as a headless job might be able to continue even if the app is suspended.

**Comment by vikeri** [comment]
You mean keeping the app alive? That's not supported. I think you'll have to start a foregroundService in order for that to work.

**Comment by cristianocca** [comment]
Yes, I'm guessing a foreground service is what I'm looking for.

It would be great for the library to support spawning a foreground service or even call the same background code as a foreground service so you can run it both while the app is running and while the app is closed.

**Comment by vikeri** [comment]
Yeah that would be great! Unfortunately I don't have the time to work on that since it's not part of how my company uses this library at the moment. PR's are very welcome 😄
